src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -329,6 +331,35 @@ private Object readResolve() {
             return getDisplayName();
         }
 
+        @Restricted(NoExternalUse.class) // for Jelly
+        public @CheckForNull String getEnclosingLabel() throws Exception {
+            FlowNode executorStepNode = context.get(FlowNode.class);
+            if (executorStepNode != null) {
+                for (StepExecution exec : executorStepNode.getExecution().getCurrentExecutions(true).get(1, TimeUnit.MINUTES)) {
```

Comment:
If we expect immediate return in all cases but errors,  wouldn't it make sense to reduce the timeout duration to something fairly short (like under 1s)?  Many users are likely to see this UI element rendered at once, so 1 min is long enough that we could still exhaust thread pools. 

I'm assuming that in the event of the timeout we're fine with just letting the TimeoutException bubble up, if it's a rare event, so not as concerned about that part.